substrate-processor/src/processor.ts

```diff
@@ -1,350 +1,640 @@
-        this.hooks.post.push({handler, ...options})
+    this.hooks.pre.push({ handler, ...options });
+  }
+
+  /**
+   * Adds a {@link BlockHandler} function that should be executed after processing each block. Can accept an 
+   * additional {@link BlockHookOptions} parameter.
+   *
+   * @param fn an asynchronous function that implements the {@link BlockHandler} interface, by accepting a 
+   * {@link BlockHandlerContext} as argument and returning a `Promise<void>`
+   */
+  addPostHook(fn: BlockHandler): void;
+  /**
+   * Adds a {@link BlockHandler} function that should be executed after processing each block.
+   *
+   * @param options an object corresponding to the {@link BlockHookOptions} interface, setting the {@link Range} of 
+   * execution for the `fn` function
+   * @param fn an asynchronous function that implements the {@link BlockHandler} interface, by accepting a 
+   * {@link BlockHandlerContext} as argument and returning a `Promise<void>`
+   */
+  addPostHook(options: BlockHookOptions, fn: BlockHandler): void;
+  /**
+   * Adds a {@link BlockHandler} function that should be executed after processing each block.
+   *
+   * @param fnOrOptions either a function implementing the {@link BlockHandler} interface, or an object corresponding 
+   * to the {@link BlockHookOptions} interface
+   * @param fn an asynchronous function that implements the {@link BlockHandler} interface, by accepting a 
+   * {@link BlockHandlerContext} as argument and returning a `Promise<void>`
+   */
+  addPostHook(
+    fnOrOptions: BlockHandler | BlockHookOptions,
+    fn?: BlockHandler
+  ): void {
+    this.assertNotRunning();
+    let handler: BlockHandler;
+    let options: BlockHookOptions = {};
+    if (typeof fnOrOptions == "function") {
+      handler = fnOrOptions;
+    } else {
+      handler = assertNotNull(fn);
+      options = fnOrOptions;
     }
-
-    addEventHandler(eventName: QualifiedName, fn: EventHandler): void
-    addEventHandler(eventName: QualifiedName, options: EventHandlerOptions, fn: EventHandler): void
-    addEventHandler(eventName: QualifiedName, fnOrOptions: EventHandlerOptions | EventHandler, fn?: EventHandler): void {
-        this.assertNotRunning()
-        let handler: EventHandler
-        let options: EventHandlerOptions = {}
-        if (typeof fnOrOptions === 'function') {
-            handler = fnOrOptions
-        } else {
-            handler = assertNotNull(fn)
-            options = fnOrOptions
-        }
-        this.hooks.event.push({
-            event: eventName,
-            handler,
-            ...options
-        })
+    this.hooks.post.push({ handler, ...options });
+  }
+
+  /**
+   * Adds an {@link EventHandler} function that should be executed when an Event with the specified name is 
+   * encountered. Can accept an additional {@link EventHandlerOptions} parameter.
+   * 
+   * @param eventName a {@link QualifiedName} (a string), representing the name of the Event of interest. When such 
```

Comment:
It should be specified what a QualifiedName is, that is, `${module}.{name}`. Should it be camel-cased?